Source/Streamstone/EntityOperation.cs

```diff
@@ -52,15 +53,18 @@ protected override TableOperation AsTableOperation()
 
             public override EntityOperation Merge(EntityOperation other)
             {
-                if (other is Insert) 
+                if (other is Insert)
                     throw InvalidMerge(other);
-                
-                if (other is Replace) 
+
+                if (other is Replace)
                     return new Insert(other.Entity);
 
-                if (other is Delete) 
+                if (other is Delete)
                     return None;
 
+                if (other is InsertOrMerge)
```

Comment:
One check per operation type seems too verbose, especially when the result is the same. It's just my opinion regarding readability...